test/distributed/fsdp/test_fsdp_comm_hooks.py

```diff
@@ -286,6 +291,82 @@ def test_registering_hook_submodules(
                 DummyHook.dummy_hook
             )
 
+    def _check_low_precision_hook(self, state, hook, sharding_strategy, dtype, has_wrapping):
+        # keep everything deterministic for input data
+        torch.manual_seed(0)
+        torch.cuda.manual_seed(0)
+
+        fsdp_with_hook = self._init_model(
+            Net(has_wrapping=has_wrapping, sharding_strategy=sharding_strategy),
+            sharding_strategy=sharding_strategy
+        )
+        fsdp_with_hook.register_comm_hook(state, hook)
+
+        mp_only_grad = MixedPrecision(reduce_dtype=dtype)
+        fsdp_with_mp = self._init_model(
+            Net(has_wrapping=has_wrapping, sharding_strategy=sharding_strategy, mixed_precision=mp_only_grad),
+            sharding_strategy=sharding_strategy,
+            mixed_precision=mp_only_grad
+        )
+
+        optim_hook = torch.optim.SGD(fsdp_with_hook.parameters(), lr=0.1)
+        optim_mp = torch.optim.SGD(fsdp_with_mp.parameters(), lr=0.1)
+
+        in_data = torch.rand(16, 8).cuda()
+        fsdp_with_hook.train()
+        fsdp_with_mp.train()
+        optim_mp.zero_grad()
+        optim_hook.zero_grad()
```

Comment:
I don't think zero_grad does anything right, since we haven't accumulated any grads yet?